build.psm1

```diff
@@ -211,6 +211,10 @@ function Start-PSBuild {
         [switch]$NoPSModuleRestore,
```

Comment:
@rjmholt, your last commit had 4 failures in `PowerShell-CI-macos`
Experimental Feature: && and || operators - Feature-Enabled.Uses the correct error semantics with statement '1,3,4,5 | Test-PipelineTerminatingError -ErrorAction SilentlyContinue -ErrorVariable nterrs || Write-Output 2'
```powershell
PIPELINE
at <ScriptBlock>, : line 1
1: 1,3,4,5 | Test-PipelineTerminatingError -ErrorAction SilentlyContinue -ErrorVariable nterrs || Write-Output 2
```
Experimental Feature: && and || operators - Feature-Enabled.Uses the correct error semantics with statement 'Test-FullyTerminatingError || Write-Output 2'
```powershell
TERMINATE
at <ScriptBlock>, /Users/vsts/agent/2.155.1/work/1/s/test/powershell/Language/Operators/PipelineChainOperator.Tests.ps1: line 63
63:             throw 'TERMINATE'

```
Experimental Feature: && and || operators - Feature-Enabled.Uses the correct error semantics with statement '1,2,3 | Test-NonTerminatingError -ErrorAction Stop || Write-Output 4'
```powershell
Expected strings to be the same, but they were different.
Expected length: 7
Actual length:   32
Strings differ at index 7.
Expected: 'NTERROR'
But was:  'NTERROR,Test-NonTerminatingError'
at <ScriptBlock>, /Users/vsts/agent/2.155.1/work/1/s/test/powershell/Language/Operators/PipelineChainOperator.Tests.ps1: line 189
189:             $_.FullyQualifiedErrorId | Should -Be $ThrownError
```
Experimental Feature: && and || operators - Feature-Enabled.Uses the correct error semantics with statement '1,3,4,5 | Test-PipelineTerminatingError -ErrorAction Stop || Write-Output 2'
```powershell
Expected strings to be the same, but they were different.
Expected length: 8
Actual length:   38
Strings differ at index 8.
Expected: 'PIPELINE'
But was:  'PIPELINE,Test-PipelineTerminatingError'
at <ScriptBlock>, /Users/vsts/agent/2.155.1/work/1/s/test/powershell/Language/Operators/PipelineChainOperator.Tests.ps1: line 189
189:             $_.FullyQualifiedErrorId | Should -Be $ThrownError
```